### PR TITLE
Assorted improvements to README.md

### PR DESCRIPTION
**Pull request by seabass-labrax**
- Add email address for subscribing to the SPDX Legal Team mailing list

- Replace 'bi-weekly' with 'fortnightly' to avoid potential confusion

- Use UTC for time of the conference calls

- Add hyperlink to CONTRIBUTING.md

- Change certain phrasing to improve clarity

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,11 +22,11 @@ This repository contains the XML source and schema files used to generate the au
...
-* **this Github repo**: comments and issues and PRs related to specific changes to the files that comprise the SPDX License List, e.g., new licenses, updates to an existing license, improving documentation, or other changes. 
+* **mailing list**: Please introduce yourself and let us know a bit about your interest in SPDX! The mailing list is our traditional form of communication. To join the SPDX Legal Team mailing list, send an email to `Spdx-legal+subscribe@lists.spdx.org` or visit https://lists.spdx.org/g/Spdx-legal (where you can also see the list's archives).
+* **fortnightly calls**: We use fortnightly conference calls to make decisions on topics and issues that may be difficult to discuss only via email or GitHub. These calls are every other Thursday at 16:00 UTC. Information, including a link to join online, is sent prior to the calls to the SPDX Legal Team mailing list. Meeting minutes for the calls are in the [SPDX meetings repo](https://github.com/spdx/meetings/tree/master/legal); minutes from meetings before March 2020 can be found at http://wiki.spdx.org/
```
Comment (zvr) [inline README.md:26]: I think the meeting time is US-based, not UTC-based. The UTC time changes when US enters/leaves Daylight Savings Time.

**Review by zvr — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (seabass-labrax) [reply README.md:26]: Thank you for pointing this out to me! So it is not just us in Europe who have to cope with changing the clocks 😀 Speaking of time zones, it's a little late here, so I'll set this to draft status and work on it tomorrow.

Comment (seabass-labrax) [reply README.md]: I've added another commit to the pull request to go back to using US Eastern Time. Thanks again!

**Comment by seabass-labrax** [comment]
In case you weren't notified by GitHub with the inline comments, I have added another commit to go back to using US Eastern Time as you suggested. Please re-review this when you have time, @zvr :)

**Comment by jlovejoy** [comment]
thanks, looks good.

**Pull request merged**